### ISSUE TITLE
genie_python: break up waitfor and change

### ISSUE THREAD

**Issue by mattclarke** [ISISComputingGroup/IBEX#1216]
There are catch-all waitfor and change commands which can be used for a number of things.

For example: waitfor(uamps=100) or waitfor(seconds=10)

From the point of view of NICOS would it be better to have separate commands, like waitfor_uamps and waitfor_time?

**Comment by ThomasLohnert** [comment]
Whilst doing this, I noticed something about the way waiting for frames and uamps works in the code:
According to documentation, the way it should work is that the program waits until at least x good frames / x uamps have been received. 
The way it actually works is that the script waits until total frame x / total uamps x have been received, e.g. waitfor_frame(3000) doesn't wait for the next 3000 good frames, but it waits for frame no. 3000 (meaning if it is already past that when the command is called, it does not wait at all)

Furthermore, I noticed the GUI only shows the last few digits of the total uamps, whereas the python method requires what I assume is the full number, so while testing instead of writing waitfor_uamps(575) corresponding to the number shown in the GUI, I had to write waitfor_uamps(35575) - this could be very confusing as I could not even see the full number anywhere (found it by trial and error).
